website/src/components/Visualisations/Visualisations.tsx

```diff
@@ -1,6 +1,154 @@
-export { default as Game } from "./Game/Game";
-export { default as Resources } from "./Resources/Resources";
-export { default as Roles } from "./Roles/Roles";
-export { default as IIGO } from "./IIGO/IIGO";
-export { default as IITO } from "./IITO/IITO";
-export { default as IIFO } from "./IIFO/IIFO";
+import React, { useEffect, useState } from "react"
+import VisualisationsNavbar from "./VisualisationsNavbar"
+import { Button, Alert } from "react-bootstrap"
+import { useHistory, Route, Switch } from "react-router-dom";
+import { gamevisualisation, visualisations, iifovisualisation, iigovisualisation, iitovisualisation, resourcesvisualisation, rolesvisualisation } from "../../consts/paths";
+import { OutputJSONType } from "../../consts/types";
+import { GitHash } from "../../consts/info";
+import { initialLoadingState, useLoadingState } from "../../contexts/loadingState";
+import { loadLocalVisOutput, clearLocalVisOutput, storeLocalVisOutput } from "./utils";
+import Game from './Game/Game'
+import IIFO from './IIFO/IIFO'
+import IITO from './IITO/IITO'
+import IIGO from './IIGO/IIGO'
+import Resources from './Resources/Resources'
+import Roles from './Roles/Roles'
+
+const Visualisations = () => {
+  const [output, setOutput] = useState<OutputJSONType | undefined>(undefined)
+  const [, setLoading] = useLoadingState()
+  const history = useHistory()
+  const [error, setError] = useState<string | undefined>(undefined)
+  const [warning, setWarning] = useState<string | undefined>(undefined)
+
+  useEffect(() => {
+    onDidMount()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  const onDidMount = async () => {
+    setLoading({ loading: true, loadingText: `We're hard at work!` })
+    try {
+      const o = await loadLocalVisOutput()
+      if (o) {
+        setOutput(o)
+        history.push(gamevisualisation)
+      }
+    }
+    catch (err) {
+      // if error, just assume not stored at all
+      console.error(err)
+    }
+    setLoading(initialLoadingState)
+  }
+
+  useEffect(() => {
```

Comment:
Do we need 2 separate calls to `useEffect`?

Reply:
Or is having multiple calls still normal practice?

Reply:
note the sensitivity list of `useEffect`--the above one is empty and means it runs on component first mount. The second one has `[output]`--it fires everytime output is changed.